src/allmydata/test/web/test_web.py

```diff
@@ -604,6 +609,15 @@ class Web(WebMixin, WebErrorMixin, testutil.StallMixin, testutil.ReallyEqualMixi
     def test_create(self):
         pass
 
+    def test_welcome_json(self):
+        d = self.GET("/?t=json")
+        def _check(res):
+            decoded = simplejson.loads(res)
+            self.failUnless("introducers" in decoded.keys())
+            self.failUnless("servers" in decoded.keys())
+        d.addCallback(_check)
```

Comment:
We should examine the response more carefully here: make sure all the keys in `["servers"]` are present and contain the right values. Does the dummy client we use for this include any servers? If not, we'll need to add one.